### PR TITLE
708 replace fuzzywuzzy and textdistance with rapidfuzz for plain evaluation metrics

### PR DESCRIPTION
**Pull request by prvenk**
In this PR, we modify the following:
- fuzz token set ratio from fuzzywuzzy with the much faster rapidfuzz version. we also make this more flexible with an option to measure other kinds of fuzz metrics, namely: ratio, partial ratio, token_sort_ratio, token_sort_partial_ratio, and token_set_partial_ratio.
- we replace some plain metrics from textdistance with the rapidfuzz version  (this is faster as documented in the issue). the metrics we replace are lccseq, hamming, jarowinkler, levenshtein. We retain textdistance variants for cosine and jaccard given rapidfuzz doesn't have a version.
- Added rouge scores as plain string metrics
- Renaming variables for clarity, e.g., doc1 to str1 and value1 to str1 given all inputs are supposed to be strings and these were not consistent.
- Adding types for arguments and return.
- Editing documentation to reflect the changes
- Added tests

### CODE REVIEW

**Comment by martinpeck** [comment]
Sorry...I don't think I have the background context to review this PR properly.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (julia-meshcheryakova) [inline rag_experiment_accelerator/evaluation/plain_metrics.py]: Can we use full names instead of d and alg for better readability? (otherwise looks good)

File: rag_experiment_accelerator/evaluation/eval.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/eval.py:65]: Severity: Minor - The change from 'FuzzyWuzzy' to 'RapidFuzz' is a good update as RapidFuzz is generally faster and more efficient than FuzzyWuzzy. However, it would be helpful to update the docstring to reflect this change more clearly, perhaps by mentioning the performance benefits of RapidFuzz over FuzzyWuzzy.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:9]: Severity: Major - The type hints for the `predictions` and `references` parameters are incorrect. They should be `list[str]` instead of `list`.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:27]: Severity: Major - The return type hint should be `float` instead of `int` for the `fuzzy` function to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:45]: Severity: Major - The return type in the docstring should be `float` instead of `list` for the `fuzzy` function to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:59]: Severity: Major - The `levenshtein` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:74]: Severity: Major - The `jaccard` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:89]: Severity: Major - The `hamming` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:104]: Severity: Major - The `jaro_winkler` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:119]: Severity: Major - The `cosine` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:152]: Severity: Major - The `lcsstr` function's return type hint should be `float` instead of `int` to match the actual return type.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/eval.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/eval.py:65]: Severity: Major - Good change! The switch from 'FuzzyWuzzy' to 'RapidFuzz' is beneficial. RapidFuzz is faster and more efficient than FuzzyWuzzy, especially for large datasets.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:45]: Severity: Major - The return type in the docstring should be `float` instead of `list`, as the function returns a single float value, not a list.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/eval.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/eval.py:84]: Severity: Major - Ensure that all parts of the codebase and dependencies are updated to use RapidFuzz instead of FuzzyWuzzy. This includes updating any import statements, function calls, and related documentation. This will prevent potential runtime errors and maintain consistency throughout the project.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

Comment (quovadim) [inline rag_experiment_accelerator/evaluation/eval.py:84]: Severity: Potential Issue - Consider updating the function's docstring to provide context for future maintainers. Mention that RapidFuzz is being used as it is faster and has fewer dependencies compared to FuzzyWuzzy. This helps maintainers understand why this library was chosen.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:29]: Severity: Potential Issue - Consider adding a check to ensure 'predictions' and 'references' lists are of equal length to avoid potential errors in BLEU score computation.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:34]: Severity: Potential Issue - The function name 'fuzzy' is not self-explanatory. Consider renaming it to something more descriptive like 'calculate_fuzzy_score' to improve readability.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:52]: Severity: Potential Issue - The docstring for the 'fuzzy' function should mention that it returns a single similarity score, not a list. Update the 'Returns' section in the docstring to 'float'.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:141]: Severity: Major - Consider adding a check to ensure 'str1' and 'str2' have the same length since the Hamming distance requires equal-length strings. Raise a ValueError if they do not match.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:29]: Severity: Potential Issue - Consider adding input validation for all functions to ensure the inputs are of the expected types and formats. For example, check that 'predictions' and 'references' are lists of strings in the 'bleu' function.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py]: Severity: Potential Issue - The use of 'getattr' to dynamically select the fuzzy matching function based on 'match_type' is a good practice, but consider validating 'match_type' against a predefined set of acceptable values to enhance robustness.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py:111]: Severity: Potential Issue - Ensure that the updated function signatures for 'levenshtein', 'jaccard', 'hamming', 'jaro_winkler', 'cosine', 'lcsseq', and 'lcsstr' are consistent across the codebase for readability and maintainability.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/plain_metrics.py]: Severity: Potential Issue - In the 'cosine' function, 'str1' and 'str2' are described as lists in the docstring, which could be misleading since they should likely be token-frequency vectors. Clarify the expected input types in the docstring.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py]: Severity: Potential Issue - The explicit cast to `int` might not be necessary if `fuzzy` already returns an integer. If it does not, consider handling the type conversion within the `fuzzy` function itself for better encapsulation.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py
Comment (quovadim) [inline rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py:24]: Severity: Potential Issue - The addition of the `match_type` argument introduces a new behavior to the `fuzzy` function. Consider adding a separate test function for different `match_type` values to isolate tests and make debugging easier.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py:29]: done

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py:34]: done

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py:52]: done

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py:111]: done

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py]: done

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py:141]: added the pad variable and set to True to account for different length strings. Cant expect the two strings coming in to be of the same length.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py]: removed the cosine function from this file as all other functions here take in strings or list of strings to compute simple metrics. cosine assumes there needs to be some encoding of strings to a vector space which we are doing in the transforner based metrics anyway.

Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py]: Tag @ritesh-modi

File: rag_experiment_accelerator/evaluation/eval.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/eval.py:84]: done

Comment (prvenk) [reply rag_experiment_accelerator/evaluation/eval.py:84]: done

File: rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py]: actual test here is to validate float rather than int. will switch to comapring apprximate values rather than casting to int

File: rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py:24]: added

File: rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/tests/test_plain_metrics.py]: done

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (kcortinas) [reply rag_experiment_accelerator/evaluation/plain_metrics.py]: The cosine function implemented here is not the traditional implementation used with embeddings, but rather, the Ochiai coefficient (see [here](https://en.wikipedia.org/wiki/Cosine_similarity)). It takes in two lists of strings so I don't think the renaming of the variables is misleading. I'd suggest to keep the metric and maybe update the name of the metric to avoid confusion (e.g. "cosine_ochiai")

File: README.md
Comment (kcortinas) [inline README.md]: If we remove or rename the cosine metric, we should update the documentation here

File: config.sample.json
Comment (kcortinas) [inline config.sample.json]: If we remove the cosine metric, we should also remove it from the sample config

File: docs/evaluation-metrics.md
Comment (kcortinas) [inline docs/evaluation-metrics.md]: Suggest to replace "But the core idea is to compare..." with "The core idea is to compare..."

File: docs/evaluation-metrics.md
Comment (kcortinas) [inline docs/evaluation-metrics.md:121]: To maintain consistency, would it be possible to summarize these metrics using a table format detailing the configuration key, calculation base and possible values?

File: docs/evaluation-metrics.md
Comment (kcortinas) [inline docs/evaluation-metrics.md]: You're referring to contexts in these definitions, but I think these metrics are computed using the actual (ground truth answer taken as reference) and expected (generated answer to the question)

File: .github/workflows/config.json
Comment (kcortinas) [inline .github/workflows/config.json:77]: Should we also test here the new ROUGE metrics added?

**Review by kcortinas — COMMENTED** [review]
Thanks @prvenk! Added a few comments and suggestions.

File: rag_experiment_accelerator/evaluation/plain_metrics.py
Comment (prvenk) [reply rag_experiment_accelerator/evaluation/plain_metrics.py]: ah ok. will retain and rename the cosine metric then. between the signature and variable name present earlier, it was unclear what it was measuring

File: .github/workflows/config.json
Comment (prvenk) [reply .github/workflows/config.json:77]: will add

File: docs/evaluation-metrics.md
Comment (prvenk) [reply docs/evaluation-metrics.md:121]: done

**Review by prvenk — COMMENTED** [review]
done

**Review by kcortinas — APPROVED** [review]
LGTM!

**Comment by prvenk** [comment]
@ritesh-modi will merge this PR once the build error on the agent resolves

**Pull request merged**
